### DISCUSSION TITLE
Disable push notifications for container updates

### DISCUSSION THREAD

**Discussion by LMAR6** [nextcloud/all-in-one#1487] [Questions] [answered]
Hi everyone, 

I really appreciate the speed of development of Nextcloud AIO. But lately there have been too many push notifications for container updates so I miss notifications from daily business. Is there any way to disable them? Or to limit them to email notifications (which I could filter)?

In the NC notifications (/settings/user/notifications) I have no possibility to limit these notifications.

Cheers!

**Comment by szaimen** [comment]
Honestly, I don't understand how these notifications which are sent on a daily basis and not every hour or something let you miss notifications from your daily business... But I suppose this is not topic of this thread.

There is indeed no way to disable them however one workaround is already mentioned in the notification: just enable daily backups which also allows to automatically update all containers which should limit the notifications to a manageable amount.

However the best approach would honestly be not using an admin account in your daily business. From a security perspective, it is much better to just use a "normal" account in your daily business and only use an admin account when you are required to do admin stuff. The main advantage is also, that "normal" user don't get these notifications.

**Reply by charredchar** [reply]
I am sorry to disagree but I do not believe this is a good solution. Even if you aren't using an admin account for your daily use the notifications are annoying. There is currently no option to modify the notification at all and getting an email once a day is extremely bothersome as I practically *always* get a daily email.

Some people do not like such applications like NextCloud to update constantly or auto-update as it opens the possibility for something major to break. I usually do updates once a month for my systems and applications and that is only after some time passes on all of them to prevent bugs from breaking things. This is pretty common for stuff like Windows Updates and NextCloud is no exception.

I personally feel like the best solution is options for AIO update notifications for the following;
Enable/Disable either/both the browser and email notification when an update is available. Just like any other notification.
Frequency of such notifications. Twice a day (this seems to be the default), once a day, once a month, only once per update. Not needed but this could help someone not just completely turn off notifications and not realize there are updates.

**Reply by szaimen** [reply]
So did you try switching to a "normal" user?

**Reply by charredchar** [reply]
Logging in as a non-admin account doesn't do anything about the constant emails. This was requested a couple of weeks ago (#1502 ) as well.

**Reply by szaimen** [reply]
> Logging in as a non-admin account doesn't do anything about the constant emails. This was requested a couple of weeks ago (#1502 ) as well.

Why not simply disable email notifications then in the user settings of your admin?

**Reply by charredchar** [reply]
Because that isn't a great idea to have zero notifications of updates, but there can be a middle ground between *way too many* and none.

Also, I've tried locating the email option and I can't find it.

![Screenshot_20230103-054342](https://user-images.githubusercontent.com/1829271/210351170-07ad7c80-a656-42dc-acc1-db1d4fe018cf.jpg)

**Reply by szaimen** [reply]
> Also, I've tried locating the email option and I can't find it.

It should be the `send email reminders about unhandled notifications` option. You can also set it to e.g. `1 week` or to `Never`.

**Reply by charredchar** [reply]
That option is in two locations, one under the personal settings and one under the admin settings. My admin settings have always been set to `Never` but I receive them every morning at 01:00 CST.

![image](https://user-images.githubusercontent.com/1829271/221304472-83e161b7-2c82-4e2b-ba52-0ad187045dcf.png)

I don't feel like it is a great idea that it is link to the unhandled notifications, either. I don't want to be forced to disable all notification alerts just because of this one really annoying one. It should, at a minimum, be added as a checkbox to enable or disable. I highly recommend, along with that, a separate option for just these as mentioned before for frequency. It is a bad idea to disable update notifications all together but tying them to normal notification frequency is a serious issue.

**Reply by ghost** [reply]
@szaimen Thank you for taking part in this discussion. Please seriously consider charredchar's request. I have the same issue and I wonder how others deal with it. Admins get nudged by AIO to continuously update the containers. Does this make sense? I don't get why these notifications exist in the first place, because admins can estimate the release cycle by themselves

> I don't want to be forced to disable all notification alerts just because of this one really annoying one. It should, at a minimum, be added as a checkbox to enable or disable.

**Comment by szaimen** [comment]
On a sidenote: how many update notifications did you get? It should only be two each day. Can you post a screenshot maybe?

**Reply by popoff1998** [reply]
I agree with charredchar. Users must have the "freedom to use" and the "freedom how to use" the software. All of our users, admins and normal, receive push notifications about "AIO is outdated". Even worse is that they receive the notification in windows & phone, so many uninstalled the client and stop using nextcloud ...

**Comment by szaimen** [comment]
Hi, we are not going to make this configurable but with https://github.com/nextcloud/all-in-one/pull/2118 container update notifications will only be sent on saturdays.

**Reply by szaimen** [reply]
This is now released with v4.6.0 Beta. Testing and feedback is welcome! See https://github.com/nextcloud/all-in-one#how-to-switch-the-channel

**Reply by ghost** [reply]
What is the rationale behind this change? Why Saturday? Why isn't it configurable?

**Comment by Cioxfisi** [comment]
Same Problem here, all our customers received the notification: "AIO is outdated" which is not fine when you have more customers and you want to do the update, by yourself and test it first.

**Reply by ecker00** [reply]
This is also plaguing our instance, had quite a lot of members ask about it.

**Comment by Snuupy** [comment]
Looking at the code, you could probably patch https://github.com/nextcloud/all-in-one/blob/main/php/src/Docker/DockerActionManager.php#L1021 to always return `false` but obviously this will come with the security implications of possibly having an outdated image which may/will be vulnerable to exploits.

**Comment by szaimen** [comment, marked as answer]
btw, we have this documentation now on the topic: https://github.com/nextcloud/all-in-one?tab=readme-ov-file#how-often-are-update-notifications-sent

**Comment by mrbrdo** [comment]
I find it mind boggling there is no setting for this. It should at least be in the iOS app settings. It's so annoying!
There is no logic in this being a (non-optional) push notification on the CLIENT when it concerns the SERVER. No other app that I ever used did it in this way. Fine if you want to make it a mandatory email or something but the push notification is too obtrusive. I only enable push notifications for the most important stuff, and this is not one of them.

What makes it even more annoying is that it's 3 push notifications for basically the same thing (Mastercontainer update available!, Container updates available!, AIO is outdated!). It's just really annoying and I hope a setting will be added to disable this completely. This should be a single email notification and not more than once a month unless there is a critical security issue.

**Comment by xaknick** [comment]
Three years later, and this annoying but trivial problem still hasn't been fixed. I also cannot understand why it cannot be configured at least through env variables. Every Saturday, I receive two notifications on each device: my iPhone, iPad, Linux laptop, and Linux desktop. If that's not considered annoying behavior, then I don't know what is.

**Comment by charredchar** [comment]
This isn't "answered" at all just by adding notes to a document stating how often it sends a notification.
Recently I've had a problem with this as the latest version is broken for me. I do not have the time to deal with it right now so I am forced to stay on an older version. But AIO seems to think I have infinite time and is now sending me notifications *every single day* about updating.
What is truly unacceptable about this is that ***NON-ADMINS*** are receiving this notification now telling them to alert the admin. There are ***no*** excuses to alert general users to nag their admins for updating.
There needs to be a setting to stop this, **now**. I don't mind an initial message on a new version (I would argue against this as well as if an admin wants such a notification they can do it through GitHub) but getting constant notifications to update when I can't without breaking my entire install for a day or longer is not how this should be forced onto admins.

**Comment by cradle8810** [comment]
The simple answer I expected is the option to stop such notifications every day.
The latest version of Nextcloud's compatible is insufficient for my computing environment at least. This is why I stayed with an older version. I want to say about this pushing the outdated notice every day "problem" is forcing update whether its compatible for another components.

I use it in the production environment too. Not a homelab. If you're using it at home, it is a good option to notify it every day for security. But it is not the same for a production environment, like an office.

Why do you push such administrative messages to admins and users every day??
